be/src/exec/pipeline/hashjoin/spillable_hash_join_probe_operator.cpp

```diff
@@ -346,64 +319,103 @@ StatusOr<ChunkPtr> SpillableHashJoinProbeOperator::pull_chunk(RuntimeState* stat
         }
 #endif
     }
+}
 
-    if (_processing_partitions.empty()) {
-        _acquire_next_partitions();
+Status SpillableHashJoinProbeOperator::_restore_probe_partition(RuntimeState* state) {
+    for (size_t i = 0; i < _probers.size(); ++i) {
+        // probe partition has been processed
+        if (_probe_read_eofs[i]) continue;
+        RETURN_IF_ERROR(_current_reader[i]->trigger_restore(
+                state, *_executor,
+                spill::ResourceMemTrackerGuard(tls_mem_tracker, state->query_ctx()->weak_from_this(),
+                                               std::weak_ptr(_current_reader[i]))));
+        if (_current_reader[i]->has_output_data()) {
+            auto chunk_st = _current_reader[i]->restore(
```

Comment:
what is the difference between trigger_restore and restore?

Reply:
trigger_restore means submit a IO task to IO executor. 
restore means get a chunk from buffer and submit a new task